app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,7 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
```

Comment:
- [ ] I think the index action of the PostsContoller should be fetching all posts for a user and kindly resolve N+1 problem by fetching all posts and comments for a user at once: `@posts = @user.posts.includes(:comments)`